ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala

```diff
@@ -64,7 +65,9 @@ object ReadServiceStateUpdateComparison {
 
   private def compareUpdates(expectedUpdate: Update, normalizedUpdate: Update): Future[Unit] = {
     val expectedNormalizedUpdate = normalizeUpdate(expectedUpdate)
-    val actualNormalizedUpdate = normalizeUpdate(normalizedUpdate)
+    // We overwrite the record time of the update with the expected one because it can vary.
```

Comment:
```suggestion
    // We ignore the record time set later by post-execution because it's unimportant, as we only care about the update type and content.
```